streams/dshr_strdata_mod.F90

```diff
@@ -1255,7 +1255,7 @@ subroutine shr_strdata_readLBUB(sdat, ns, mDate, mSec, newData, istr, rc)
     call ESMF_TraceRegionExit(trim(istr)//'_setup')
 
     ! if model current date is outside of model lower or upper bound - find the stream bounds
-    if (rDateM < rDateLB .or. rDateM > rDateUB) then
+    if (rDateM < rDateLB .or. rDateM >= rDateUB) then
```

Comment:
I've convinced myself this change is correct. Because, it will update the data when rDateM == rDateUB. So this should be done in preparation of that.